File: types/quiz.ts
export enum QuestionType {
  SINGLE_CHOICE = "SINGLE_CHOICE",
  MULTIPLE_CHOICE = "MULTIPLE_CHOICE",
  TRUE_FALSE = "TRUE_FALSE",
}

export interface QuizOption {
  id: string
  text: string
  isCorrect?: boolean // Only visible when reviewing
}

export interface QuizQuestion {
  id: string
  text: string
  imageUrl?: string
  type: QuestionType
  options: QuizOption[]
  explanation?: string
}

export interface Quiz {
  id: string
  title: string
  description?: string
  createdBy: string
  createdAt: string
  updatedAt?: string
  isPublic: boolean
  tags: string[]
  timeLimit: number // in minutes, 0 means no time limit
  questions: QuizQuestion[]
}

export interface QuizSubmission {
  answers: Record<string, string[]> // questionId -> selected optionIds
  timeSpent: number // in seconds
}

export interface QuizAttempt {
  id: string
  quizId: string
  quizTitle: string
  score: number
  totalQuestions: number
  correctAnswers: number
  wrongAnswers: number
  unanswered: number
  timeSpent: number // in seconds
  completed: boolean
  startedAt: string
  completedAt?: string
}

File: backend/src/main/java/com/quizmaster/model/Quiz.java
package com.quizmaster.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "quizzes")
public class Quiz {
    
    @Id
    private String id;
    
    @Indexed
    private String title;
    
    private String description;
    
    private List<String> tags;
    
    private String createdBy;
    
    private boolean isPublic;
    
    private int timeLimit; // in minutes, 0 means no time limit
    
    private List<Question> questions;
    
    @CreatedDate
    private LocalDateTime createdAt;
    
    @LastModifiedDate
    private LocalDateTime updatedAt;
    
    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Question {
        private String id;
        private String text;
        private String imageUrl;
        private QuestionType type;
        private List<Option> options;
        private String explanation;
        
        public enum QuestionType {
            SINGLE_CHOICE, MULTIPLE_CHOICE, TRUE_FALSE
        }
        
        @Data
        @Builder
        @NoArgsConstructor
        @AllArgsConstructor
        public static class Option {
            private String id;
            private String text;
            private boolean isCorrect;
        }
    }
}
